pkg/services/publicdashboards/metric/metric.go

```diff
@@ -0,0 +1,74 @@
+package metric
+
+import (
+	"context"
+	"errors"
+	"strconv"
+	"time"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/publicdashboards"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type PublicDashboardsMetricServiceImpl struct {
```

Comment:
is it strictly necessary? I think you normally create an interface in order to implement it in many places, such as mocks, decorators and so on. I don't see it necessary

Reply:
It's not, but the name implies that. Could we change the name to just `Service` in that case? Since it is in its own package `metric.Service` it sounds good.

Reply:
you mean PublicDashboardsMetricService?

Reply:
I'm not a fan of adding such big prefixes to anything so I was suggesting `Service`. This Metric Service seems to be called only from the background_service to initialize it and we can rename the package when importing if we want to be more explicit when referencing the service. But I don't have a strong opinion on this one.